scipy/interpolate/_aaa.py

```diff
@@ -309,6 +319,72 @@ def _compute_weights(self, f, z, rtol, max_terms):
         self.support_values = fj[i_non_zero]
         self.weights = wj[i_non_zero]
         self.errors = errors[: m + 1]
+        self._points = z
+        self._values = f
+
+    def clean_up(self, cleanup_tol=1e-13):
+        """Automatic removal of Froissart doublets.
+
+        Parameters
+        ----------
+        cleanup_tol : float, optional
+            Poles with residues less than this number times the geometric mean
+            of `values` times the minimum distance to `points` are deemed spurious by
+            the cleanup procedure, defaults to 1e-13.
+        """
+        # Find negligible residues
+        if any(self._values):
```

Comment:
What does `any(self._values)` mean here, what are you actually checking?

```
In [13]: any(np.array([]))
Out[13]: False

In [14]: any(np.array([0]))
Out[14]: False

In [15]: any(np.array([1]))
Out[15]: True

In [16]: any(np.array([0, 0]))
Out[16]: False

In [19]: any(np.array([0.]))
Out[19]: False
```

Reply:
Turns out this want needed by letting `gmean` handle everything